src/gl.rs

```diff
@@ -60,6 +60,7 @@ fn bpp(format: GLenum, pixel_type: GLenum) -> GLsizei {
         ffi::UNSIGNED_SHORT => 2,
         ffi::SHORT => 2,
         ffi::FLOAT => 4,
+        ffi::UNSIGNED_INT_8_8_8_8_REV => 4,
```

Comment:
this should be 1, actually. Since it's meant to return the size of a *component*, and 4 is the size of the whole pixel.
The code then goes and multiplies it by the number of components, so with the current code it would return 16 for BGRA instead of 4

Reply:
Yeah good catch. I think I'll restructure it more.